source/particle/property/interface.cc

```diff
@@ -192,6 +192,10 @@ namespace aspect
               }
           }
 
+        AssertThrow(particle_properties.size() == n_property_components,
+                    ExcMessage("The wrong number of properties were mapped onto "
+                               "a newly generated particle."));
```

Comment:
I think you might as well make this an `ExcInternalError`. This should simply not happen if I understand you correctly.